### PR TITLE
Fixed that SSO failed with avatar in Unicode URL

### PR DESCRIPTION
**Pull request by henrynet**
Added encoding unicode strings in avatar_url to prevent getting bad URI error in Discourse.

### CODE REVIEW

File: lib/discourse.php [hunk 1]
```diff
@@ -200,10 +200,10 @@ function sso_parse_request( $wp )
   function get_avatar_url( $user_id ) {
     $avatar = get_avatar( $user_id );
     $doc = new DOMDocument();
-    $doc->loadHTML( $avatar );
+    $doc->loadHTML( mb_convert_encoding( $avatar, 'HTML-ENTITIES', 'UTF-8' ) );
     $xpath = new DOMXPath( $doc );
     $src = $xpath->evaluate( 'string(//img/@src)' );
-    return $src;
+    return utf8_uri_encode($src);
```
Comment (retlehs) [inline lib/discourse.php]: can you please update this line to be consistent with the other formatting? (space after opening paren, space before closing paren)

**Comment by swalkinshaw** [comment]
@henrynet while you're here, would you updating the function to use `preg_match` like WP core does instead of this crazy DOM manipulation?

See https://github.com/WordPress/WordPress/blob/cd0ba24e9583a707b0ba055f0a3d9cd0f9b36549/wp-admin/includes/misc.php#L734

**Comment by henrynet** [comment]
@swalkinshaw I have tested to get the avatar url by preg_match( "|src='([^']+)'|", $avatar, $matches ) and failed. Because the get_avatar function puts the src within " not ' so the match will be null.

ref: http://wordpress.stackexchange.com/questions/59442/how-do-i-get-the-avatar-url-instead-of-an-html-img-tag-when-using-get-avatar

**Comment by henrynet** [comment]
@retlehs, @swalkinshaw I have updated my commit cb4e4622bbe23efcdd370804ffd656d38e03e6c7, please check if there are any other problems.
Thank you so much.

File: lib/discourse.php [hunk 2]
```diff
@@ -199,11 +199,9 @@ function sso_parse_request( $wp )
...
   function get_avatar_url( $user_id ) {
     $avatar = get_avatar( $user_id );
-    $doc = new DOMDocument();
-    $doc->loadHTML( $avatar );
-    $xpath = new DOMXPath( $doc );
-    $src = $xpath->evaluate( 'string(//img/@src)' );
-    return $src;
+    if( preg_match( "/src=['\"](.*?)['\"]/i", $avatar, $matches ) )
+      return utf8_uri_encode( $matches[1] );;
+    return NULL;
```
Comment (swalkinshaw) [inline lib/discourse.php]: No need for `return NULL;` since PHP handles that on its own.

There's also 2 semicolons on the line above

**Comment by swalkinshaw** [comment]
@henrynet thanks :+1:. Looks much better other than my comment.

**Comment by henrynet** [comment]
@swalkinshaw thanks for the comments.
I have created a new PR #118 for updating get_avatar_url with all fixes.

**Pull request closed without merging**